tools/roslaunch/src/roslaunch/loader.py

```diff
@@ -501,6 +501,17 @@ def param_value(self, verbose, name, ptype, value, textfile, binfile, command):
                     # special handle the use of Python scripts in Windows environment:
                     # 1. search for a wrapper executable (of the same name) under the same directory with stat.S_IXUSR flag
                     # 2. if no wrapper is present, prepend command with 'python' executable
+                    def split_with_single_quote_enclosed_fixup(command):
+                        tokens = shlex.split(command, posix=False)  # use non-posix method on Windows
+                        if not ("'" in command):
+                            return tokens
+                        new_command = []
+                        for token in tokens:
+                            if token.startswith("'") and token.endswith("'"):
+                                new_command.append(token[1:-1])
+                            else:
+                                new_command.append(token)
```

Comment:
This loop could be expressed with a list comprehension which would make it more readable.